docs: improve UDF signature docs (#8194)
ibis-project/ibis: docs/how-to/extending/builtin.qmd

con.execute(cardinality(["a", "b"]))
```

When you do not specify input types, Ibis isn't able to catch typing errors
early, and they are only caught during execution.
The errors you get back are backend dependent:

```{python}
#| error: true
con.execute(cardinality("foo"))
```

> Here, Duckdb is informing us that the `ARRAY_LENGTH` function does not accept

Reviewer: ```suggestion
Here, DuckDB is informing us that the `ARRAY_LENGTH` function does not accept
```

con.execute(cardinality(["a", "b"]))
```

> When you skip giving input types, Ibis isn't able to catch your typing errors

Reviewer: ```suggestion
When you do not specify input types, Ibis isn't able to catch typing errors
```
